Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/ConfigConsole/ConfigConsoleController.cs

```diff
@@ -1,23 +1,33 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
 namespace Dnn.PersonaBar.ConfigConsole.Components
 {
     using System;
     using System.Collections.Generic;
     using System.IO;
     using System.Linq;
+    using System.Threading;
     using System.Xml;
-
+
     using DotNetNuke.Application;
     using DotNetNuke.Common;
-    using DotNetNuke.Common.Utilities;
+    using DotNetNuke.Common.Utilities;
+    using DotNetNuke.Instrumentation;
 
     public class ConfigConsoleController
     {
+        private static readonly ILog Logger = LoggerSource.Instance.GetLogger(typeof(ConfigConsoleController));
+        private const string CONFIG_EXT = ".config";
+        private const string ROBOTS_EXT = "robots.txt";  // in multi-portal instances, there may be multiple robots.txt files (e.g., site1.com.robots.txt, site2.com.robots.txt, etc.)
+
         public IEnumerable<string> GetConfigFilesList()
         {
-            var files = Directory.GetFiles(Globals.ApplicationMapPath, "*.config");
+            var files = Directory
```

Comment:
I'm not sure how more efficient that would be since the number of files would be low in almost every use case and we're just pulling the two file types.

Reply:
We can leave it, just an observation